test/common/config/utility_test.cc

```diff
@@ -295,6 +296,65 @@ TEST(UtilityTest, PrepareDnsRefreshStrategy) {
   }
 }
 
+void packTypedStructIntoAny(ProtobufWkt::Any& typed_config, const ProtobufWkt::Message& inner) {
+  udpa::type::v1::TypedStruct typed_struct;
+  (*typed_struct.mutable_type_url()) =
+      std::string("type.googleapis.com/") + inner.GetDescriptor()->full_name();
+  MessageUtil::jsonConvert(inner, *typed_struct.mutable_value());
+  typed_config.PackFrom(typed_struct);
+}
+
+TEST(UtilityTest, TypedStructToStruct) {
+  ProtobufWkt::Any typed_config;
+  ProtobufWkt::Struct untyped_struct;
+  (*untyped_struct.mutable_fields())["foo"].set_string_value("bar");
+  packTypedStructIntoAny(typed_config, untyped_struct);
+
+  ProtobufWkt::Struct out;
+  Utility::translateOpaqueConfig(typed_config, ProtobufWkt::Struct(),
+                                 ProtobufMessage::getStrictValidationVisitor(), out);
+
+  EXPECT_TRUE(Protobuf::util::MessageDifferencer::Equals(out, untyped_struct));
```

Comment:
Use this pattern instead here and in other tests:
`EXPECT_THAT(config.runtime(), ProtoEq(bootstrap.layered_runtime()));`

Reply:
done